app/controllers/api/red_hat_migration_analytics_controller.rb

```diff
@@ -46,18 +64,35 @@ def find_provider_ids(type)
     end
 
     class << self
-      def parse_manifest
-        default_manifest_path = Cfme::MigrationAnalytics::Engine.root.join("config", "default-manifest.json")
-        user_manifest_path    = Pathname.new("/opt/rh/cfme-migration_analytics/manifest.json")
-        manifest_path         = user_manifest_path.exist? ? user_manifest_path : default_manifest_path
 
-        manifest = Vmdb::Settings.filter_passwords!(load_manifest(manifest_path))
+      def default_manifest_path
+        @default_manifest_path ||= Cfme::MigrationAnalytics::Engine.root.join("config", "default-manifest.json")
+      end
+
+      def user_manifest_path
+        @user_manifest_path ||= Pathname.new("/opt/rh/cfme-migration_analytics/manifest.json")
+      end
 
+      def store_manifest(manifest)
+        File.open(user_manifest_path, 'w') do |f|
+          f.write(JSON.generate(manifest))
+        end
```

Comment:
Minor, but simpler:

```ruby
File.write(user_manifest_path, JSON.generate(manifest))
```

Reply:
Actually, if it's a Pathname, you can just do

```ruby
user_manifest_path.write(JSON.generate(manifest))
```

Reply:
Thanks, @Fryguy!

Fixed.